feature_engine/encoding/base_encoder.py

```diff
@@ -101,7 +98,7 @@ def _check_fit_input_and_variables(self, X: pd.DataFrame) -> pd.DataFrame:
         # save train set shape
         self.n_features_in_ = X.shape[1]
 
-        return X
+        return self
```

Comment:
The typehint and docstring should be changed to match this updated return behavior.  I might suggest not returning anything, but I did see @solegalli suggested specifically `self`.

Reply:
OK fixed in latest commit. Convention in rest of code for `return self` seems to be blank typehint and no mention in docstring, so I went with that.

Reply:
@solegalli  OK should I change to not return `self`?